### DISCUSSION TITLE
リリースサイクルとバージョンタグ

### DISCUSSION THREAD

**Discussion by kazzyfrog** [first-contributions-ja/first-contributions-ja.github.io#164] [【運営】相談・報告]
バージョンの付け方というか、刻み方に関する相談です！

今まで通りmainブランチ＋作業用ブランチでいく方向であれば、
リリースサイクルとバージョンタグに関しては、
以下のような形がいいかも？と考えたのでメモしておきます！

- ▶︎n.9：機能追加は一旦終えて、**仕様の確認**作業や修正を行う
- →n.0：リリース & 機能追加はせず、**仕様には変更がない**コードやサイトの設定系の改善を行う
- →n.1：次のバージョンのリリースに向けて機能実装
- →n.9：・・・

上記でいくと、現在は、v0.9！
リリースしたら、v1.0.0
リファクタリング系終わったらv1.1.0
次のリリースまでの機能実装して、v1.9.0

つまり、マイナーバージョン（v1.2.3の2。真ん中の数字）は、
9 or 0 or 1 で、リリース前、リリース後、機能実装を判別する的な！？

**Comment by kazzyfrog** [comment]
GitHubの、リリースノート作成画面での説明：

**Tagging suggestions**
It’s common practice to prefix your version names with the letter v. Some good tag names might be v1.0.0 or v2.3.4.

If the tag isn’t meant for production use, add a pre-release version after the version name. Some good pre-release versions might be v0.2.0-alpha or v5.9-beta.3.

**Semantic versioning**
If you’re new to releasing software, we highly recommend to [learn more about semantic versioning.](http://semver.org/)

A newly published release will automatically be labeled as the latest release for this repository.

If 'Set as the latest release' is unchecked, the latest release will be determined by higher semantic version and creation date. [Learn more about release settings.](https://docs.github.com/repositories/releasing-projects-on-github/managing-releases-in-a-repository)

**Reply by kazzyfrog** [reply]
訳：
**タグ付けの提案**
バージョン名の前に文字vを付けるのが一般的です。いくつかの良いタグ名は、v1.0.0またはv2.3.4かもしれません。

タグが本番使用用でない場合は、バージョン名の後にプレリリース版を追加します。いくつかの良いプレリリースバージョンは、v0.2.0-alphaまたはv5.9-beta.3かもしれません。

**セマンティックバージョニング**

ソフトウェアのリリースが初めての場合は、セマンティックバージョニングについて詳しく学ぶことを強くお勧めします。
新しく公開されたリリースは、このリポジトリの最新リリースとして自動的にラベル付けされます。

「最新のリリースとして設定」がチェックされていない場合、最新のリリースはより高いセマンティックバージョンと作成日によって決定されます。リリース設定の詳細をご覧ください。

**Comment by kazzyfrog** [comment]
#163 に、以下のように書きました！
> GitHubでは、プルリクの内容に応じて、リリースノートを自動生成する機能などもあります！

作成した、バージョンタグと、リリースノートは、すぐに消したり編集したりできます！

なので、一旦練習で、
v0.9のバージョンタグとリリースノートを、作成（自動生成）してみた方がわかりやすいですかね？

**Reply by kazzyfrog** [reply]
その場合、手順のスクショも撮るつもりです！

**Reply by kagomen** [reply]
早速ありがとうございます！
自分のリポジトリで練習してみて、なんとなく流れを把握しました。

> 一旦練習で、
v0.9のバージョンタグとリリースノートを、作成（自動生成）してみた方がわかりやすいですかね？

ですね！
実際どんな感じで反映されるか見たいです。！

あとはyamlファイルでラベル別に分類できるみたいですが……
個人的には流れが追いづらくなりそうだから必要ないかなーという気もしますが、どうなんでしょうか👀

**Comment by pss-aileen** [comment]
すみません、全然理解できていないので
まともに議論できないのですが
私の脳みそはこんな感じでやると良いのかな...？と考えています。

簡単に言うと、リリース前はbetaとして、本リリース？ローンチの時にbetaをはずす感じはどうかな...と

| version | 内容 |
|---|---|
| v0.0.0<br>or v1.0.0-beta | 今 |
| v1.0.0 | リリース |
| v1.0.1 | 誤字やドキュメントなど軽微な修正 |
| v1.1.0 | 内部的なリファクタリング？ある程度のもの |
| v1.1.1 | 誤字やドキュメントなど軽微な修正 |
| v2.0.0-beta | バージョン2の実装 |
| v2.0.0 | バージョン2 リリース |
| v2.0.1 | 修正第1弾 |
| v2.0.2 | 修正第2弾 |
| v2.0.3 | 修正第3弾... |
| v2.1.0 | themeの追加（例！！） |

参考
- [正しいバージョンの付け方を身につけよう | 株式会社LYZON](https://www.lyzon.co.jp/blog/2019/20191017_how_to_attach_version_name/)
- [セマンティックバージョニングによるバージョン表記方法](https://zenn.dev/mo_ri_regen/articles/semantic-versioning)
- [セマンティック バージョニング 2.0.0 |セマンティック バージョニング](https://semver.org/)
- [ソフトウェアのバージョンと開発スタイルの関係について | フューチャー技術ブログ](https://future-architect.github.io/articles/20220426a/)

**Reply by kazzyfrog** [reply]
ほぼほぼ同じ認識です！

自分が書いたのは、
betaを、１つ前バージョンのバージョンのマイナーを「9」で表すことで、
もうすぐメジャーバージョンの数字が上がる！的なことを、見ただけで簡単に表したかっただけなので、
「v2.0.0-beta」の形でも、全然良いと思います！

リリース前を、「v2.0.0-beta」とするか、「v1.9.0」とするか、という違いですね！

**Comment by kazzyfrog** [comment]
https://github.com/first-contributions-ja/first-contributions-ja.github.io/releases

試しに練習で、
v0.9のバージョンタグとリリースノートを、作成（自動生成）してみました！
あとで消したり編集したりもできます！👍

特に自分で変更したりすることなく、自動生成のままです！
プルリクのタイトルが全部入っている感じ！

**Reply by kazzyfrog** [reply]
こういう感じでよければ、これをベースに色々決めていければと思ってます！

下記など！
- バージョンの付け方
- リリースノートの内容も色々カスタマイズできる！→確か、fix, chore、、などのprefix毎にまとめるとかもできたはず、、。

**Reply by kagomen** [reply]
ありがとうございますー！

リリースノートのカスタマイズについてなんですが、上で、

> あとはyamlファイルでラベル別に分類できるみたいですが……
個人的には流れが追いづらくなりそうだから必要ないかなーという気もしますが、どうなんでしょうか

と書いたのですが、実際見てみるとプルリクが思ったより多いので、ラベル別の分類した方が良いかも？と思いました。
ただ、ラベル別の分類をするとやはり流れが追いづらくなるかなとも思うので、
ラベル別の分類をするのであれば、ラベルの付け方について、今一度再検討したいなと思いました。

今あるprefixは

- feat
- fix
- style
- docs
- chore

ですが、個人的にはこの中で、`fix`と`feat`の使いどころ？が難しいと感じています。
例えば、

- スタイルの修正は`fix`なのか`style`なのか
- レスポンシブデザインの実装は`feat`なのか`style`なのか

といった具合です。

リリースノートでも変更の流れが追えるようにするのであれば、`feat`や`fix`は削除して、

- スタイル関連は全部style
- docs関連は全部docs
- 設定関連は全部chore

として、`style` `docs` `chore`の3つだけにするのはどうかな？と考えたのですが、どうでしょうか？

**Reply by pss-aileen** [reply]
おっしゃっている通り
- docsは修正であれ、追加であれdocs
- 設定関連は修正であれ、追加であれ設定

とするのが良いのかなと思いました。

最近prefixの概念？というか2つのグループに分類できるなと思って。

1. コードにかかわるもの、実装にかかわるものは以下の**コードのprefix**。
2. docs、初期設定やインストール系は、**コード以外のprefix**。

そういった感じにわけるとやりやすいかなと思っています。

### コード
- `feat`: 今まで実装していなかったもの、新しく実装。スタイル設定も新しく書くならこちら
- `fix` or `improve` 的なもの: 実装済みのものをアップデート。スタイル設定のアップデート、修正。
- `bugFix`: エラーが吐かれるもの、致命的なものを修正する

### コード以外のもの
- `docs`
- `chore`

**Reply by kagomen** [reply]
今気づきましたが、`style` `docs` `chore` だけだと、リファクタリングとかは何に分類するんだって感じですね💧

だったら`style`をなくして、Aileenさんのおっしゃる通り`feat`と`fix` にコード系をまとめた方が良いかもと思いました！
（それとも、`style`は残す方向ですか？）

あと、`bugFix`の導入は賛成ですー！！！
ずっと`fix` に 🐛=bug の絵文字がつくのが個人的にはしっくりこなくて、もし見解一致した場合は絵文字も合わせて変更できたら良いなと思います

リリースノート的にも`bugfix`欄があると、バグ修正されたんだ！ということが一目でわかって良さそうですよね！

**Reply by pss-aileen** [reply]
自分が想定しているprefix一覧は
- feat
- fix(or improve)
- bugFix
- docs
- chore

です！styleは新しくスタイル加えるのであればfeat、すでに作ったスタイルを修正する時はfixでよいかな〜と思っています。
ただ、featとfixってちょっと難しい感じがするので迷いはあるかもしれませんが
とにかく新しく、今までそこになかったものを作る場合はfeat、今あるコードからの修正リファクタリングはfixかなと。
（伝わりますか...？😭）

bugFixも私も悩んでて
やっぱりbugFixと何かの修正は分けたほうが良いな〜と思っていたので、bugFix入れる方向でいけたらと思います！

**Reply by kazzyfrog** [reply]
現状こんな感じですかね？
- feat：新規追加系に関すること（機能、UIセクションやコンポーネントなど、）
- docs：ドキュメントに関すること全般（作成、修正、追加など、）
- chore：設定に関すること全般（設定ファイル作成、変更、削除、追加など、）

個人的に、以下の使い分けがいまいちピンときてないです。。
- fix(or improve)：
- bugFix

多分、バグの修正とそれ以外のスタイル修正やリファクタとかですかね、？

であれば、docsやchoreと同じ流れで、こうでもいいかも！
- fix：バグの修正
- style：バグではない、スタイル（見た目）の変更
- :recycle:  refactor：仕様に影響がない（機能追加でもバグ修正でもない）コード改善・リファクタ

**Reply by kagomen** [reply]
@kazzyfrog 

### 疑問
- 今までのfixは全部refactorになる、という認識で合ってますか？👀（今までバグの修正をしたことがないと思うので）
- style関連のrefactor(やfix)は、styleにする感じですか？👀
- 自分の予想だと以下のような感じなのかな？と思ったのですが、合ってますか❓

ソースコード全般
　 ├ ✨ feat: 新機能追加
　 ├ 🐛 fix: バグ修正
　 └ ♻️ refactor: コード改善
🎨 style: スタイル関連
📚 docs: ドキュメント関連
🔧 chore: 設定関連

**Reply by kazzyfrog** [reply]
> 今までのfixは全部refactorになる、という認識で合ってますか？👀（今までバグの修正をしたことがないと思うので）

仕様に関する事というか、直さないといけないことは、fix（バグ修正）でいい気がしてます！
例えば、apple iconとか、OG画像が表示されないとか、、は正しい形に直さないといけないので、不具合と言えそうなので、fix。
一方、リファクタは、最悪変更しなくてもいいというか、ユーザーがサイトを使う分には関係ない変更って感じですかね！

> style関連のrefactor(やfix)は、styleにする感じですか？

style を使う場合は、
docsやchoreと同じように、style関連の変更は、styleでいいのかなと思いました！

> 自分の予想だと以下のような感じなのかな？と思ったのですが、合ってますか❓

あってます！
個人的には、fixとbugFixの違いが曖昧になりそうかも、、と思ったので、 ♻️ refactorを使うという提案でした！

**Reply by kagomen** [reply]
> apple iconとか、OG画像

こちらはなんとなく `chore` なのかな？と思っていたのですが、 `fix` になるのでしょうか？

> style を使う場合は、
docsやchoreと同じように、style関連の変更は、styleでいいのかなと思いました！

`style` を入れるなら、これはそうした方が良いですよね。！

**Reply by kazzyfrog** [reply]
> こちらはなんとなく chore なのかな？と思っていたのですが、 fix になるのでしょうか？

この質問の意図としては、
「サイトのOG画像やapple iconが正しく設定されていない」ことが、不具合かどうか、という質問でしょうか？
もしくは、不具合だけど、設定に関することなので、choreの方が良いのでは、という質問でしょうか？

個人的には、不具合だけど、設定に関することなので、その場合fix（バグ修正）でいいと思ってました！🙌

**Reply by kazzyfrog** [reply]
featは、新規機能やUIコンポーネントなどの追加

多分ややこしいのが、変更時のprefixですよね！！

- バグ（不具合）の修正は、fix
- バグではない変更の場合
  - ドキュメント関連は、docs
  - 設定（ライブラリ、設定ファイル、メタデータ、ビルドやCI、など）は、chore
  - （機能追加でもバグ修正でもない）コード改善・リファクタは、refactor
  - （もしこれを入れるなら）UIや見た目の変更・改善は、style

こんな感じなのかな。。🤔

**Reply by kagomen** [reply]
> 不具合だけど、設定に関することなので、choreの方が良いのでは

こちらです！

リリースノートをラベル（カテゴリー）別で分ける際は、

- コード全般（fix/feat/refactor）
- スタイル関連（style）
- 設定関連（chore）
- ドキュメント関連（docs）

以上の4つに分けるのだろうと考えており、
その場合、例えばOGPのバグ修正をfixにしてしまうと、OGPの設定は設定関連に記載されるのに、バグ修正はコード関連に記載されたりして、機能追加や修正の流れが追いづらくなるのでは？と考えていました。

また、OGPのバグ修正がfixになるのであればstyleのバグ修正もfixになると思います。
そういった、choreやstyleの壁を超えるくらいの、カテゴリーとしての支配力（？）がfixにはあるんでしょうか❓

伝わりますか…？

**Reply by kagomen** [reply]
> choreやstyleの壁を超えるくらいの、カテゴリーとしての支配力（？）がfixにはあるんでしょうか？

こちらあまりにもわかりくい表現かなと思い、追記です💦

![IMG_1053](https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/154225199/fe105388-abf8-436b-90c5-011b64296ee1)

多分上記のような図になってると思うんですが、この場合、↑で自分が述べた

> コード全般（fix/feat/refactor）
スタイル関連（style）
設定関連（chore）
ドキュメント関連（docs）

ではなく、

> バグ関連（fix）
スタイル関連（style）
設定関連（chore）
ドキュメント関連（docs）
その他コード全般（feat/refactor）

としてリリースノートを区分する感じになるのかな？と

**Reply by kazzyfrog** [reply]
> リリースノートをラベル（カテゴリー）別で分ける際は、
〜〜〜
OGPのバグ修正をfixにしてしまうと、OGPの設定は設定関連に記載されるのに、バグ修正はコード関連に記載されたりして、機能追加や修正の流れが追いづらくなるのでは？と考えていました。

確かに、その場合、バグ修正はコード関連に記載されて、少しわかりづらくなりますね！！

> choreやstyleの壁を超えるくらいの、カテゴリーとしての支配力（？）がfixにはあるんでしょうか❓

個人的には、
「機能追加と、バグ修正と、それ以外の変更（それぞれに適したラベル）」というくらいの感覚だったので、
仕様と違うこと・想定していないことの修正（「修正すべき」というより、「修正しないといけない」こと）は、
設定系であれスタイルであれ、fixを使うべきかなという認識でした！
（これは完全に個人の見解というか、認識的なことなので、違う方法で統一してもOKです！）

それと、リリースノートをラベル（カテゴリー）別で分ける際は、
prefixのラベルは全部並列で、下記のような形でいいのでは？と思いました！
（コード全般でまとめる必要はないかも？と思った次第です。。！！）
参考までに！

---

## What's Changed
### ✨ New Features
* A追加
### 🐛 Bug Fix
* Bの不具合を修正
### 📚 Documentation
* C追記
### 🎨 Style
* スタイルDの適用
### ♻️ Refactoring
* Eをコンポーネントに分割
### 🔧 Chore
* Fの導入
### Other Changes
* その他

**Reply by kagomen** [reply]
> 「機能追加と、バグ修正と、それ以外の変更（それぞれに適したラベル）」というくらいの感覚だった

なるほど！ということは、条件分岐の仕方として、まず

**機能追加か？バグ修正か？それ以外か？**

があって、次に

**スタイル関連か？設定関連か？ドキュメント関連か？**

があるんですね！（合ってますか？）

**変更の流れを追う**ことを優先させるのであれば、
スタイルはバグ修正も変更も全部styleに、とした方が良いかなと思いますが、

**機能追加やバグ修正の告知**を優先させるのであれば

> prefixのラベルは全部並列

が適してるなと思いました！

メンテナー視点で（というか、リリースノートというものに対して）、変更の流れを追うことを第一優先に考えてましたが、ユーザー視点に立つと、割とそんな変更の流れはどうでもよくて、機能追加やバグ修正の告知の方が大事かも！と思いました。

ならば、今のprefix（＋ refactor を追加？ ）のままでも問題ないな、と！

**Reply by kazzyfrog** [reply]
> （合ってますか？）

おっしゃる通りです！
個人的には、機能追加とバグ修正が、他よりかは重要度が１つ高いような感覚でした！！

> ユーザー視点に立つと、割とそんな変更の流れはどうでもよくて、機能追加やバグ修正の告知の方が大事かも！と思いました。

そうですね！！
そもそも、リリースノート自体、何十回も継続的にリリースされていくようなアプリやライブラリなどが前提で、
「このバージョンリリースで、こういう機能が追加されて、こういうバグが解消されたんだ〜〜！」
って感じの記録というか告知だと思うので、リリースノート内での流れはそんなに重要じゃないかな？と思ってます！

リリースごとの流れを記録する役割が強いのかな、と思ってます！👍

**Reply by kazzyfrog** [reply]
問題なければ、 [上の例](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/164#discussioncomment-9020550)に掲載したような形で、
一度.github/release.yml ファイルを作成してみますよ！

それをみて、以下など？を確認・検討してみてもいいかもです！
- styleやrefactorは必要か
- fixとbugFixの方が良いか
- コード関連と、その他でまとめた方がいいか
- 過去のPRのラベル付けが適切か

**Reply by kagomen** [reply]
はい！大丈夫だと思います！
ありがとうございます……！🙏✨

> それをみて、以下など？を確認・検討してみてもいいかもです！

了解です！！

**Comment by kazzyfrog** [comment]
メモ：
このスレッドに関しては、以下を決めれば、イシュー( #163 )片付きそうです！

- バージョンタグの付け方（リリース前を、「v2.0.0-beta」とするか、「v1.9.0」とするか）
- バージョンは、v1.0.0のリリースから付け始めるか、現在（「v1.0.0-beta」or「v0.9.0」）から付けるか
- リリースノートは、[この通り](https://github.com/first-contributions-ja/first-contributions-ja.github.io/releases)自動生成だと全部並列なので、カスタマイズするかどうか。

**Reply by kazzyfrog** [reply]
個人的には、
- わかりやすので、「v2.0.0-beta」の表記で良さそう
- せっかくなので、「v1.0.0-beta」を最初につけても良さそう
- 一旦、自動生成のままでいいかも？

**Reply by kazzyfrog** [reply]
あ、でもせっかくなら、リリースノートのカスタマイズもやってみた方がいいかもです
Githubの色々を使い倒すのも１つのテーマ？なので！

**Reply by pss-aileen** [reply]
- v0.9.0よりかは、v1.0.0-betaのほうが、一般的に？わかりやすいかな？と思ったので、v1.0.0-betaが良いかなと思いました !
- リリースノートのカスタマイズはちょっと時間がないので手が出せませんが分別できると良いな〜と思いました！

**Reply by kazzyfrog** [reply]
そしたら以下で良さそうですかね？

- バージョンタグの付け方は、[この通り](https://github.com/first-contributions-ja/first-contributions-ja.github.io/discussions/164#discussioncomment-8972005)
  - リリース前を「v1.0.0-beta」：機能追加は一旦終えて、仕様の確認作業や修正を行っている
  - リリース後は「v1.0.0」： リリースして、機能追加はせず、仕様には変更がないコードやサイトの設定系の改善を行っている
  - 機能の実装は「v1.1.0」：次のバージョンのリリースに向けて機能実装している
  - バグの修正などは、適宜パッチ（v1.2.3の3。１番右の数字）を上げていく
- バージョンは、現在の「v1.0.0-beta」から付け始める
- リリースノートは、上記のprefixのスレッドでの内容で、カスタマイズする！
  - カスタマイズは、プルリクについたラベルに基づいて、自動で生成できるようになる：[参考：GitHub のリリースノート自動生成機能を使う](https://developer.mamezou-tech.com/blogs/2022/03/11/github-automatically-generated-release-notes/)
  - 現状は、その通りにprefixをつけられていないので、手書きで見出しをつけて編集する！（もしくは、過去のPR全部にラベルつける？→半分くらい変更なさそうなのはつけた！）
  - 確かこれ、Github Actionsとかで、PRに自動でラベルつけることできたはず。。

**Reply by kazzyfrog** [reply]
> バージョンは、現在の「v1.0.0-beta」から付け始める

と思ったのですが、
v1.0.0-beta（ドッグフーディングを行なった）時点では、
.github/release.yml ファイルが作成されていないので、自動生成リリースノートをラベル毎に分けて生成することができませんでした！
全部がカテゴリラベルなしで、並んでいる状態、、。

4/6、 #167 にて、
.github/release.yml ファイルを作成したので、このコミット以降であれば、
自動生成リリースノートをラベル毎に分けて生成することができます！

なので、「v1.0.0」からのリリースでいいかもと思いました！
（それであれば、.github/release.yml ファイルのコミットも含まれるので、問題なくラベル毎に生成できます！）

**Reply by kazzyfrog** [reply]
問題なければ、
「v1.0.0-beta」のタグを削除して、「v1.0.0」を最初のタグとして作成します！！

---

（追記）
「v1.0.0-beta」のタグは削除しなくても、「v1.0.0-beta」のリリースノートを作成しなければ、タグ自体は残せました！

まとめ：
- v1.0.0-beta（ドッグフーディングを行なった）時点の、バージョンタグは残せるので残す
- v1.0.0-beta時点では、リリースノートのカスタマイズ（ラベル毎に分けて自動生成する）が適用されないので、リリースノートは生成しない
- v1.0.0では、リリースノートのカスタマイズ（ラベル毎に分けて自動生成する）が適用されるので、リリースノートを生成する！

**Comment by kazzyfrog** [comment]
## バージョンタグとリリースノートの作成手順

**Reply by kazzyfrog** [reply]
## 1. リポジトリ右側のサイドバーの、「Releases」もしくは、「Create a new release」をクリック！
<img width="372" alt="スクリーンショット 2024-04-06 3 56 33" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/17518dd2-a362-447a-a8cd-0f5c5ac97ac2">

**Reply by kazzyfrog** [reply]
## 2. 下記の作成画面になる
<img width="955" alt="スクリーンショット 2024-04-06 4 00 37" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/2731da7c-98e4-409e-b81a-ea1f73a674c7">

**Reply by kazzyfrog** [reply]
## 2-1 .Choose a tagで、バージョンタグを選択 or 作成できるので、「v1.0.0」のバージョンタグを作成する
<img width="357" alt="スクリーンショット 2024-04-06 4 01 47" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/f328df03-b2ad-4867-9b15-99cf8c15c24d">

**Reply by kazzyfrog** [reply]
## 2-2. バージョンタグを選択したら、上部のメニューが少し変わり、「Generate release notes」がアクティブになるので、クリックする！
<img width="956" alt="スクリーンショット 2024-04-06 4 04 07" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/d7b0078e-6c68-44ac-bf8b-60b040aa05d9">

**Reply by kazzyfrog** [reply]
## 2-3. .github/release.ymlの設定にそって、自動でリリースノートの内容が生成されていることを確認する！

- ここで、自由に編集もできる！
- .github/release.ymlが、設定されていない場合、そのバージョンのプルリクエスト（タイトル＋コントリビューター名）が、すべて列挙される感じ！

<img width="952" alt="スクリーンショット 2024-04-06 4 07 24" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/7efba5d1-3d8a-49ef-9ee7-e448a3570c1d">

**Reply by kazzyfrog** [reply]
## 2-4. 必要があれば、下の２つの項目にチェックを入れて、「Publish release」をクリックする！
- 「Set as a pre-release」は、α版やβ版のリリースの際に使う。チェックされていると、リリースノートに「pre-release」と表示される！
- 「Create a discussion for this release」がチェックされていると、このリリースノートについての、ディスカッションのスレッドを立てることができる！（ディスカッションのカテゴリを指定できる）

<img width="578" alt="スクリーンショット 2024-04-06 4 14 54" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/25cab68b-1b93-46b9-8153-ddec37d7b297">

**Reply by kazzyfrog** [reply]
生成されたリリースノートがこちら：
https://github.com/first-contributions-ja/first-contributions-ja.github.io/releases

バージョンの横に、Pre-releaseと表示されている！
<img width="274" alt="スクリーンショット 2024-04-06 4 21 14" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/9968a7a7-53f4-43bb-8491-75f7627b9de4">

右下に、このリリースに関する、ディスカッションのスレッドへのリンクが貼られる！
→ #168 
<img width="987" alt="スクリーンショット 2024-04-06 4 22 13" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/9b036861-dee1-417c-9f34-ec21d9a5c997">

**Reply by kazzyfrog** [reply]
テスト終了！！⏰

**Reply by kazzyfrog** [reply]
ちなみに、バージョンタグがあれば、その時点でのソースコードを見ることができる！
リリースノートは、あくまでそのバージョンでの、内容をまとめて周知するためのノート！
（現時点では、タグは「v1.0.0-beta」「v1.0.0」の２つあるが、リリースノートは「v1.0.0」だけ）

<img width="407" alt="スクリーンショット 2024-04-06 5 42 39" src="https://github.com/first-contributions-ja/first-contributions-ja.github.io/assets/70629747/8b70e7b1-7fe6-4ac5-9326-29305d5fee18">

**Reply by kagomen** [reply]
ありがとうございます！！！
ひとまずwikiにurl貼り付けるだけ貼り付けておきました！
